exclusion optimizations: dynamic exclusions (part of #1216):

ikreymer opened the pull request:
- instead of restarting crawler when exclusion added/removed, add a message to a redis list (per crawler instance)
- no longer filtering existing queue on backend, now handled via crawler
- depends on webrecorder/browsertrix-crawler#408 PR in the crawler
- match response optimization: instead of returning first 1000 matches, limits response to 500K and returns however many matches fit in that response size (for optional pagination on frontend)

ikreymer committed:
exclusion optimizations: dynamic exclusions (part of #1216):
- instead of restarting crawler when exclusion added/removed, add a message to a redis list (per crawler instance)
- no longer filtering existing queue on backend, now handled via crawler
- depends on webrecorder/browsertrix-crawler#408 PR in the crawler
- match response optimization: instead of returning first 1000 matches, limits response to 500K
and returns however many matches fit in that response size (for optional pagination on frontend)

tw4l reviewed (approved):
Looks good!

Only observation is that this is another change where we're requiring a certain version of the crawler - not sure if we wanna formalize that in some way.

ikreymer committed:
Merge branch 'main' into dynamic-exclusions

ikreymer commented:
Should be good to go now that Crawler 0.12.1 is out!

Also adding two tweaks:
- Exclusions entered into the text box, but not added with `+` are still added when user clicks 'Done Editing' to close the window, to avoid mistakes where exclusion is entered but not added.

- There is still a delay when exclusions are being removed on the backend, while still showing up in the queue. However, can provide more immediate feedback by crossing them off in the queue display, until they are removed on a subsequent refresh:

<img width="1190" alt="Screenshot 2023-11-03 at 5 44 36 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/1015759/ca82cef8-b9e4-492b-8256-08f4f25404eb">

ikreymer committed:
exclusions ui tweaks:
- auto-add exclusions entered into form when user clicks 'Done Editing' if they forgot to click '+'
- render URLs in queue that are removed according to current regexes as strikethrough to provide immediate feedback
about removed URLs after new exclusion is added while backend is still processing removal

ikreymer committed:
ensure workflow editor correctly switches to '#watch' tab when new crawl is started after editing

ikreymer committed:
Merge branch 'main' into dynamic-exclusions

tw4l commented:
Tested and good to merge!
